### DISCUSSION TITLE
AddTabToActiveTabSet Troubleshoot

### DISCUSSION THREAD

**Discussion by johnnytranamp** [caplin/FlexLayout#359] [General]
Hi, I'm having issues implementing a button to add a new tab within my app. I've tried looking thru the demo and example code provided but can't seem to figure out what the issue is. Any help or guidance would be greatly appreciated thank you! (New Frontend Engineer)

Below I've also attached the code that I'm working along the console log message within my local host that I've been receiving every time I click the "Add" button. The console log test message seems to be working but the code breaks right after that.

<img width="955" alt="localhost" src="https://user-images.githubusercontent.com/118769555/205709235-7cf49247-3f35-4ec3-9a33-d1ca0e93c06f.PNG">
<img width="477" alt="consolelog" src="https://user-images.githubusercontent.com/118769555/205708501-502295c9-c9bd-4449-9904-7e46619277e6.PNG">
<img width="414" alt="jsonmodel" src="https://user-images.githubusercontent.com/118769555/205708502-cf73a51d-6b7c-431d-9306-71c8dec20912.PNG">
<img width="473" alt="app" src="https://user-images.githubusercontent.com/118769555/205708497-534391e8-70d1-41ef-bdfa-8bd0003af230.PNG">
<img width="549" alt="render" src="https://user-images.githubusercontent.com/118769555/205708504-8a7d8cb8-48f1-43f0-86fa-d86266be5f82.PNG">

**Comment by nealus** [comment]
looks like you haven't set the layout ref, try adding 

ref={this.layoutRef}

to your <Layout element

**Reply by johnnytranamp** [reply]
Thank you for the tip! The second console log message is now able to pass and display within the browser but I'm still not able to have/see a new tab created.

<img width="476" alt="consolelog2" src="https://user-images.githubusercontent.com/118769555/205720344-ddec4153-a920-46b6-a8d9-ab92d07fe95a.PNG">
